File: binarysearch.io/October 2020/69_level_order_traversal.cpp
#include "solution.hpp"
using namespace std;
/**
 * class Tree {
 *     public:
 *         int val;
 *         Tree *left;
 *         Tree *right;
 * };
 */
class Solution {
    public:
    vector<int> solve(Tree* root) {
        vector<int> level;
        queue<Tree *> nodes;
        if(!root) return level;
        nodes.push(root);
        while(!nodes.empty()){
            int size = nodes.size();
            while(size--){
                Tree * node = nodes.front();
                nodes.pop();
                level.push_back(node->val);
                if(node->left) nodes.push(node->left);
                if(node->right) nodes.push(node->right);
                
            }
        }
        return level;
    
    }
};

File: binarysearch.io/October 2020/69_level_order_traversal.py
# class Tree:
#     def __init__(self, val, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def solve(self, root):
        if root is None:
            return []
        level = []
        nodes = []
        nodes.append(root)
        while nodes:
            new = []
            for node in nodes:
                level.append(node.val)
                if node.left:
                    new.append(node.left)
                if node.right:
                    new.append(node.right)
            nodes = new 
        return level 
        
